src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPushThread.java

```diff
@@ -106,84 +107,118 @@ public Vulnerability apply(VulnerableComponentView vulnerableComponentView) {
                     ? "http://web.nvd.nist.gov/view/vuln/detail?vulnId=" + vulnerableComponentView.getVulnerabilityWithRemediation().getVulnerabilityName()
                     : "");
             vulnerability.setSeverity(String.valueOf(vulnerableComponentView.getVulnerabilityWithRemediation().getSeverity()));
-            vulnerability.setScanDate(bomUpdatedValueAt);
-            return vulnerability;
-        }
-    };
+            vulnerability.setScanDate(maxBomUpdatedDate);
+            vulnerabilities.add(vulnerability);
+        });
+        return vulnerabilities;
+    }
 
-    public BlackDuckFortifyPushThread(BlackDuckFortifyMapper blackDuckFortifyMapper) {
-        this.blackDuckFortifyMapper = blackDuckFortifyMapper;
+    public BlackDuckFortifyPushThread(BlackDuckFortifyMapperGroup blackDuckFortifyMapperGroup) {
+        this.blackDuckFortifyMapperGroup = blackDuckFortifyMapperGroup;
     }
 
     @Override
     public void run() {
```

Comment:
Refracted the code and now this method is having 20 lines (Commented lines are excluded)